File: src/rust/min_max_iterativo.rs
fn main() {
    min_max(vec![54,42,11,33,24,99,77,80]);
}

fn min_max(vetor: Vec<i32>) {
    let mut min = vetor[0];
    let mut max = vetor[0];

    for i in 1..vetor.len() {
        if vetor[i] < min {
            min = vetor[i];
        } else if vetor[i] > max {
            max = vetor[i];
        }
    }

    println!{"Data: {:?}", vetor};
    println!{"Min.: {:?}", min};
    println!{"Max.: {:?}", max};
}

File: src/cpp/MinMaxIterativo.cpp
#include <iostream>

using namespace std;

int main() {

    int arr[] = {1, 5, -20, 6, 15};

    int min = arr[0];
    int max = arr[0];

    for(auto& i : arr) {
        if(i < min) min = i;
        else if(i > max) max = i;
    }

    cout << "Max = " + to_string(max) + "\nMin = " + to_string(min) << endl;

    return 0;
}
